pages/studentAssessmentInstanceHomework/studentAssessmentInstanceHomework.js

```diff
@@ -40,76 +40,67 @@ router.get('/', function (req, res, next) {
   if (res.locals.assessment.type !== 'Homework') return next();
   debug('is Homework');
 
-  ensureUpToDate(res.locals, (err) => {
+  ensureUpToDate(res.locals, async (err) => {
```

Comment:
@nwalters512 I'd like to have this entire endpoint wrapped in `asyncHandler` from `express-async-handler`. However, `ensureUpToDate` calls [`update` from `assessments.js`](https://github.com/PrairieLearn/PrairieLearn/blob/ed27bcfa732e8a8e1ff6833a06499a34fa85d7bf/lib/assessment.js#L126), which employs the use of callbacks. Would it be sensible to rewrite `assessment.update` to remove the need for a callback, or is wrapping this GET endpoint in `express-async-handler` even necessary?

For context, I was able to use all async code in the `studentAssessmentInstanceExam.js` equivalent of this endpoint: https://github.com/PrairieLearn/PrairieLearn/blob/2399e8accdf4dfda3e7a2d5d8bb5792414c80602/pages/studentAssessmentInstanceExam/studentAssessmentInstanceExam.js#L104-L106

Reply:
You should be able to wrap `assessment.update` in `utils.promisify`. Here's an example:

https://github.com/PrairieLearn/PrairieLearn/blob/ed27bcfa732e8a8e1ff6833a06499a34fa85d7bf/lib/server-jobs.js#L344

Then you can make this local `ensureUpToDate` function async.